doc/proposals/dualstack.md

```diff
@@ -0,0 +1,280 @@
+# Whereabouts support for Ipv4/IPv6 DualStack
+
+# Table of contents
+
+- [Introduction](#introduction)
+  - [Goal](#goal)
+- [Design](#design)
+  - [Changes in IPAM Config](#configChanges)
+  - [Changes in Modules](#moduleChanges)
+- [Alternative Design](#alternative)
+- [Summary](#summary)
+
+<hr>
+
+## Introduction <a name="introduction"></a>
+
+Starting v1.20, Kubernetes officially started supporting DualStack, which is being used extensively in many projects. 
+
+Whereabouts being an important IPAM CNI plugin, should also make a plan for supporting DualStack.
+This is a design proposal document for introducing IPv4/IPv6 DualStack support in whereabouts.
+
+
+### Goal <a name="goal"></a>
+
+- Allocate IPv4 and/or IPv6 addresses to pods depending on IPAM configuration
+- Above feature should not break the backward compatibility for SingleStack configuration
+- Allocate multiple (even more than 2) IPs to the pods irrespective of IPv4/IPv6 _(Optional and depends on design choice)_
+
+<hr>
+
+## Design <a name="design"></a>
+
+The idea is to update the IPAM config format to include multiple IP configurations.
+
+Basically, rather than having IP related configurations directly as key members of `IPAMConfig` object, they be included as element of an object type, let's say `IPConfig` and `IPAMConfig` should have an array of `IPConfig` as it's element.
+
+### Changes in IPAM Config <a name="configChanges"></a>
+
+This will lead to change in IPAM config something as follows:
+
+<table>
+<tr>
+<th>Old IPAM Config</th>
+<th>Changes</th>
+<th>New IPAM Config</th>
+</tr>
+<tr>
+<td>
+  
+```json
+{
+      "cniVersion": "0.3.0",
+      "name": "whereaboutsexample",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "192.168.2.225/28",
+        "exclude": [
+           "192.168.2.229/30",
+           "192.168.2.236/32"
+        ]
+      }
+}
+```
+  
+</td>
+<td>
+
+```diff
+{
+      "cniVersion": "0.3.0",
+      "name": "whereaboutsexample",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "whereabouts",
+-       "range": "192.168.2.225/28",
```

Comment:
I think we both want to keep this for backwards compatibility.

Reply:
Ooops, I see this is one example, I see that other keeps this.

Reply:
Yes :)